packages/toolpad-app/src/toolpad/AppEditor/PageEditor/JsExpressionEditor.tsx

```diff
@@ -41,25 +44,45 @@ export function JsExpressionEditor({
     const type = jsonToTs(globalScope);
 
     const globals = Object.keys(globalScope)
-      .map((key) => `declare const ${key}: RootObject[${JSON.stringify(key)}];`)
-      .join('\n');
+      .map((key) => {
+        let declaration = ``;
+        const metaData = globalScopeMeta[key] || {};
```

Comment:
Could this potentially give trouble when someone passes `{ toString: 1 }` to `globalScope`? Perhaps this needs to check with [`hasOwnProperty`](https://github.com/mui/mui-toolpad/blob/594e6c7750de34029cea2604983d165e81f9e35e/packages/toolpad-app/src/utils/collections.ts#L11)

Reply:
In other instances of this component the `globalScope` is derived from user input, so we don't know what gets passed. e.g. someone could create a TextField named "toString".